index.js

```diff
@@ -3983,6 +3983,26 @@
   S.splitOn =
   def('splitOn', {}, [$.String, $.String, $.Array($.String)], splitOn);
 
+  //# splitOnRegex :: RegExp -> String -> Array String
+  //.
+  //. Returns the substrings of its second argument separated by evaluated
+  //. matches of its first argument.
```

Comment:
@jordrake my take at a better description would be something like:

> Takes a regex and a string and returns the substrings that do not match the regex.

but probably best to wait till @davidchambers can review the PR :)